Sources/ToolchainRegistry/Toolchain.swift

```diff
@@ -281,37 +293,53 @@ extension Toolchain {
         foundAny = true
       }
 
-      // If 'currentPlatform' is nil it's most likely an unknown linux flavor.
-      let dylibExt: String
-      if let dynamicLibraryExtension = Platform.current?.dynamicLibraryExtension {
-        dylibExt = dynamicLibraryExtension
-      } else {
-        logger.fault("Could not determine host OS. Falling back to using '.so' as dynamic library extension")
-        dylibExt = ".so"
+      func findDylib(named name: String, searchFramework: Bool = false) -> URL? {
+        // If 'currentPlatform' is nil it's most likely an unknown linux flavor.
+        let dylibExt: String
+        if let dynamicLibraryExtension = Platform.current?.dynamicLibraryExtension {
+          dylibExt = dynamicLibraryExtension
+        } else {
+          logger.fault("Could not determine host OS. Falling back to using '.so' as dynamic library extension")
+          dylibExt = ".so"
+        }
```

Comment:
You could pull this out so that we only run it once (and avoid the log spam if we couldn't determine the platform).

Reply:
We should only create `Toolchain`s when creating the `ToolchainRegistry`, so I’m not sure how we would pull this out further.

Reply:
I suppose it's not actually clear what I highlighted here in the UI. By "pull out" I just meant moving the `dylibExt` logic out of this function (and the loop) so that we only do it once rather than up to twice * number of search paths.

Reply:
Ah, I see. Pulled it out.